### PR TITLE
[OAI]:  Fix Anthropic Bedrock Client Bug

### LINKED ISSUE #3198: [Bug]: Bug when using anthropic bedrock
### Describe the bug

Hi, I cloned the latest version of your autogen repository and packaged it for use, to utilize the newly added Anthropics Bedrock interface. I discovered several bugs during the process, which I would like to provide for your reference:
(1) Regarding the configuration file of Anthropics Bedrock , aws_session_token is not mandatory. Therefore, in line 111 of the code located at [./autogen/autogen/oai/anthropic.py](https://github.com/microsoft/autogen/blob/b7bdbe1ecc1c00abd5172f472c6051d0230249bd/autogen/oai/anthropic.py#L111), it is unnecessary to check its value.
(2) In line 490 of [/maindata/data/user/qiaoqiao.jin/3DGPT/autogen/autogen/oai/client.py](https://github.com/microsoft/autogen/blob/b7bdbe1ecc1c00abd5172f472c6051d0230249bd/autogen/oai/client.py#L490), the input should be config instead of openai_config, as openai_config is None at that point.

### Steps to reproduce

_No response_

### Model Used

_No response_

### Expected Behavior

_No response_

### Screenshots and logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by wenngong**
Fix #3198

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by Hk669** [comment]
thanks @wenngong , was about to make a PR, made changes locally.

**Comment by wenngong** [comment]
> thanks @wenngong , was about to make a PR, made changes locally.
> can you fix the tests that are failing. for the code formatting use: `pre-commit`

Ok, I will fix it.

**Comment by wenngong** [comment]
@microsoft-github-policy-service agree

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 install_requires = [
     "openai>=1.3",
+    "anthropic>=0.23.1",
```
Comment (Hk669) [inline setup.py:22]: anthropic shouldn't be the required dependencies. it is optional, for the devs using anthropic models.

**Review by Hk669 — CHANGES_REQUESTED** [review]

**Comment by wenngong** [comment]
> > thanks @wenngong , was about to make a PR, made changes locally.
> > can you fix the tests that are failing. for the code formatting use: `pre-commit`
> 
> Ok, I will fix it.

@Hk669 I have fixed the test failure. and pre-commit passed. Please review.

image

image

File: setup.py (hunk 1 above)
Comment (Hk669) [reply setup.py:22]: instead you can update the `extra_kwargs` in the client.py and make an `anthropic_config` variable

Comment (wenngong) [reply setup.py:22]: so, for autogen, other llm clients are just dev optional? not regular choices between openai and others?

**Comment by Hk669** [comment]
> > Ok, I will fix it.
> 
> @Hk669 I have fixed the test failure. and pre-commit passed. Please review.
> 
> image image

but the anthropic, shouldnt be in the required dependencies, can you please check on that.
cc @marklysze

File: setup.py (hunk 1 above)
Comment (marklysze) [reply setup.py:22]: Hey @wenngong, just catching up on this one...

@Hk669 is right in that it's not a mandatory dependency for all AutoGen users to have anthropic's library installed. It is an optional install, e.g. `pip install pyautogen[anthropic]` which is defined in the "extra_require" section of the `setup.py`.

So we can leave the setup.py file as it was :)

Comment (wenngong) [reply setup.py:22]: > Hey @wenngong, just catching up on this one...
> 
> @Hk669 is right in that it's not a mandatory dependency for all AutoGen users to have anthropic's library installed. It is an optional install, e.g. `pip install pyautogen[anthropic]` which is defined in the "extra_require" section of the `setup.py`.
> 
> So we can leave the setup.py file as it was :)

OK，got it, I will restore setup.py file. Thanks @Hk669 @marklysze

**Comment by wenngong** [comment]
OAI_CONFIG_LIST config file example:

```
[
    {
        "model": "anthropic.claude-3-5-sonnet-20240620-v1:0",
        "aws_access_key":<accessKey>,
        "aws_secret_key":<secretKey>,
        "aws_region":"us-east-1",
        "api_type": "anthropic"
    }
]
```

**Comment by Hk669** [comment]
@wenngong this seems to be a good PR, can you please check this https://github.com/microsoft/autogen/pull/3210

**Comment by wenngong** [comment]
> @wenngong this seems to be a good PR, can you please check this #3210

@Hk669 I believe my PR is better. In my pr, the bedrock config just depends on AnthropicClient, just like what openai_kwargs did.

**Comment by wenngong** [comment]
> > @wenngong this seems to be a good PR, can you please check this #3210
> 
> @Hk669 I believe my PR is better. In my pr, the bedrock config just depends on AnthropicClient, just like what openai_kwargs did.

@Hk669 I really think do not disperse bedrock related kwargs to client.py

File: autogen/oai/client.py [hunk 2]
```diff
@@ -370,7 +370,8 @@ class OpenAIWrapper:
 
     openai_kwargs = set(inspect.getfullargspec(OpenAI.__init__).kwonlyargs)
     aopenai_kwargs = set(inspect.getfullargspec(AzureOpenAI.__init__).kwonlyargs)
-    openai_kwargs = openai_kwargs | aopenai_kwargs
+    anthropic_kwargs = set(inspect.getfullargspec(AnthropicClient.__init__).args)
+    openai_kwargs = openai_kwargs | aopenai_kwargs | anthropic_kwargs
```
Comment (sonichi) [inline autogen/oai/client.py:374]: This line is not intuitive: should openai_kwargs contain anthropic_kwargs?

Comment (Hk669) [reply autogen/oai/client.py:374]: i believe #3210 fixes this issue.

**Comment by Hk669** [comment]
@wenngong thanks for the efforts. this issue is fixed in the other PR.

**Pull request closed without merging**